### PR TITLE
Update Map Fragment

### PR DESCRIPTION
**Pull request by HoangChu-Claremont**
I have been working with ParseDB to perform basic GET and POST request. I've also learned how to put a pointer to a Database, and have a concrete procedure in mind when transitioning from JSON to DB. I've also seen a lot of refactoring possibilities as well.

Expect to complete Show Details of Each Destination, User Can Bookmark / Save Tours, and a heavy refactor tomorrow.

https://user-images.githubusercontent.com/65592963/176609973-ba762df1-8c20-4939-a138-bd007a08bada.mp4

### CODE REVIEW

File: app/src/main/java/com/example/where_to_go/fragments/HomeFragment.java [hunk 1]
```diff
@@ -60,34 +63,76 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+
+        RecyclerView rvRecentTours = requireView().findViewById(R.id.rvRecentTours);
+
+        mostRecentTours = new ArrayList<>();
+        recentPathAdapter = new FeaturedPathAdapter(getContext(), mostRecentTours);
+
+        // Set Layout Manager
+        LinearLayoutManager tLayoutManager = new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false);
+        rvRecentTours.setLayoutManager(tLayoutManager);
+        rvRecentTours.setHasFixedSize(true); // always get top 10 paths
```
Comment (kvn93) [inline app/src/main/java/com/example/where_to_go/fragments/HomeFragment.java:117]: I don't get the relation between this line and the comment. What do you mean by "always get top 10 paths"?

File: app/src/main/java/com/example/where_to_go/models/Destinations.java [hunk 2]
```diff
@@ -12,13 +12,13 @@
 import java.util.Collections;
 import java.util.List;
 
-public class Destination {
+public class Destinations {
```
Comment (kvn93) [inline app/src/main/java/com/example/where_to_go/models/Destinations.java:15]: why change it to `Destinations`? It holds a single destination right? then the name should be in singular `Destination`.

File: app/src/main/res/layout/fragment_profile.xml [hunk 3]
```diff
@@ -1,14 +1,122 @@
 <?xml version="1.0" encoding="utf-8"?>
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (kvn93) [inline app/src/main/res/layout/fragment_profile.xml:2]: If you use constraintLayout, you should not use any other ViewGroup inside.

File: app/src/main/java/com/example/where_to_go/fragments/HomeFragment.java (hunk 1 above)
Comment (HoangChu-Claremont) [reply app/src/main/java/com/example/where_to_go/fragments/HomeFragment.java:117]: I imagined the Feature Tours will display 10 featured tours maximum in Home Page. Sorry for the naming confusion, I will renamed all Path(s) to Tour(s) later.

File: app/src/main/java/com/example/where_to_go/models/Destinations.java (hunk 2 above)
Comment (HoangChu-Claremont) [reply app/src/main/java/com/example/where_to_go/models/Destinations.java:15]: I initially thought that the class name extending ParseObject should be named similarly to the DB name, in which case this is "Destinations". I will refactor it.

**Pull request merged**